docs/dotnet/dotnet-aspnet-metrics.md

```diff
@@ -0,0 +1,196 @@
+- [Routing](#routing)
+  * [Metric: `aspnet.routing.match_attempts`](#metric-aspnetroutingmatch_attempts)
+- [Exception metrics](#exception-metrics)
+  * [Metric: `aspnet.diagnostics_handler.exceptions`](#metric-aspnetdiagnostics_handlerexceptions)
+- [Rate-limiting](#rate-limiting)
+  * [Metric: `aspnet.rate_limiting.active_request_leases`](#metric-aspnetrate_limitingactive_request_leases)
+  * [Metric: `aspnet.rate_limiting.request_lease.duration`](#metric-aspnetrate_limitingrequest_leaseduration)
+  * [Metric: `aspnet.rate_limiting.queued_requests`](#metric-aspnetrate_limitingqueued_requests)
+  * [Metric: `aspnet.rate_limiting.request.time_in_queue`](#metric-aspnetrate_limitingrequesttime_in_queue)
+  * [Metric: `aspnet.rate_limiting.requests`](#metric-aspnetrate_limitingrequests)
+
+<!-- tocstop -->
+
+## Server
+
+## Routing
+
+All routing metrics are reported by `Microsoft.AspNetCore.Routing` meter.
+
+### Metric: `aspnet.routing.match_attempts`
+
+<!-- semconv metric.aspnet.routing.match_attempts(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `aspnet.routing.match_attempts` | Counter | `{match}` | Number of requests that were attempted to be matched to an endpoint. [1] |
+
+**[1]:** Meter name is `Microsoft.AspNetCore.Routing`.
+<!-- endsemconv -->
+
+<!-- semconv metric.aspnet.routing.match_attempts(full) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `aspnet.routing.match_status` | string | Match result - success or failure | `success`; `failure` | Recommended |
+| `aspnet.routing.fallback` | string | TODO | `TODO` | Required |
+| `http.route` | string | The matched route (path template in the format used by the respective server framework). See note below [1] | `/users/:userID?`; `{controller}/{action}/{id?}` | Required |
+
+**[1]:** MUST NOT be populated when this is not supported by the HTTP server framework as the route attribute should have low-cardinality and the URI path can NOT substitute it.
+SHOULD include the [application root](/docs/http/http-spans.md#http-server-definitions) if there is one.
+
+`aspnet.routing.match_status` has the following list of well-known values. If one of them applies, then the respective value MUST be used, otherwise a custom value MAY be used.
+
+| Value  | Description |
+|---|---|
+| `success` | Match succeeded |
+| `failure` | Match failed |
+<!-- endsemconv -->
+
+## Exception metrics
+
+Metrics reported by `Microsoft.AspNetCore.Diagnostics` meter.
+
+### Metric: `aspnet.diagnostics_handler.exceptions`
+
+<!-- semconv metric.aspnet.diagnostics_handler.exceptions(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `aspnet.diagnostics_handler.exceptions` | Counter | `{exception}` | Number of exceptions caught by exception handling middleware. [1] |
```

Comment:
I wonder if diagnostics_handler should just be diagnostics, e.g. aspnet.diagnostics.exceptions